External XrootD built from git commit is not detected

- [x] Checked for duplicates
<!--
Please search in
* [GitHub](https://github.com/root-project/root/issues?q=is%3Aissue)
* AND [Jira](https://sft.its.cern.ch/jira/issues/?jql=project %3D ROOT)
for existing reports of your issue.

If you find one, you are very welcome to add to the existing report, for instance "issue still exists in today's master".
-->

### Describe the bug
I am trying to build Root (6.24.00_patches) against external XrootD that was built from a commit (`332967c` in my case).
However, ROOT complains that it can't find Xrootd:
```
-- Looking for XROOTD
-- Found Xrootd version num: 20220304-332967c (setting -DROOTXRDVERS=20220304)
-- Setting OLDPACK TRUE
CMake Error at cmake/modules/SearchInstalledSoftware.cmake:908 (message):
  XROOTD not found.  Set environment variable XRDSYS to point to your XROOTD
  installation, or inlcude the installation of XROOTD in the
  CMAKE_PREFIX_PATH.  Alternatively, you can also enable the option
  'builtin_xrootd' to build XROOTD internally
Call Stack (most recent call first):
  CMakeLists.txt:219 (include)
```

### Expected behavior
XroodD should be detected

### Additional context
In my case, the version of XrootD, as defined in `XrdVersion.hh` is `20220304-332967c` (date + commit ID, see [xrootd code](https://github.com/xrootd/xrootd/blob/master/genversion.sh#L193-L196)). Since it is not a dotted version, FindXROOTD [assumes](https://github.com/root-project/root/blob/master/cmake/modules/FindXROOTD.cmake#L47) that this is some old XrootD and uses the date part as a version. 

One possible fix would be to look for `-` in version and use that as an indication of untagged version (same as if version is `unknown`, which is only used if XrootD couldn't use git to determine own version).

--- a/cmake/modules/FindXROOTD.cmake
+++ b/cmake/modules/FindXROOTD.cmake
@@ -35,7 +35,7 @@ if (XROOTD_INCLUDE_DIR)
   if (${xrdvers} STREQUAL "\"unknown\"")
     math(EXPR xrdversnum 1000000000)
   else ()
-    string(REGEX REPLACE "[^v\\.]+" "" xrdversdots ${xrdvers})
+    string(REGEX REPLACE "[^v\\.-]+" "" xrdversdots ${xrdvers})
     if (${xrdversdots} STREQUAL "v..")
        # Regular version string; parse it out
        string(REGEX MATCH "[0-9\\.]+" xrdvers ${xrdvers})
@@ -43,6 +43,9 @@ if (XROOTD_INCLUDE_DIR)
        string(REGEX REPLACE "^([^.]*)\\.(.*)\\.(.*)" "\\2" xrdversminor ${xrdvers})
        string(REGEX REPLACE "^([^.]*)\\.(.*)\\.(.*)" "\\3" xrdverspatch ${xrdvers})
        math(EXPR xrdversnum ${xrdversmajor}*100000000+${xrdversminor}*10000+${xrdverspatch})
+    elseif (${xrdversdots} STREQUAL "v-")
+       # Untagged commit
+       math(EXPR xrdversnum 1000000000)
     else ()
        # Old version string: we keep only the first numerics, i.e. the date
        string(REGEX REPLACE "[v\"]" "" xrdvers ${xrdvers})

